libs/Microsoft.MixedReality.WebRTC/DataChannel.cs

```diff
@@ -160,7 +160,12 @@ internal DataChannel(IntPtr nativeHandle, PeerConnection peer, IntPtr argsRef, i
         public void Dispose()
         {
             State = ChannelState.Closing;
+            // This will invoke some callback from native, which calls Destroy()
```

Comment:
There seem to be a few dodgy things going on here:
- the same back and forth of control between classes that's going on in tracks/transceivers. It would be simpler if `PeerConnection.RemoveDataChannel` simply _removed_ the DataChannel from the PC and the call to `Destroy` were explicit (here and in the PC).
- the fact that a disposable class can be destructed under the hood without an explicit call to Dispose is a bit weird (though acceptable as long as it's documented)
- `Dispose()` should be safe to call twice - especially if it can be called under the hood without the user's request.

Reply:
Fixed. `DataChannel` is not `IDisposable` anymore. `PeerConnection` owns a collection of them to keep the C# objects alive while a native implementation is alive too, between the add and remove calls.